### ISSUE TITLE
Attempting to access secrets in initializers or environment fails

### ISSUE THREAD

**Issue by codyrobbins** [rails/rails#19853]
After upgrading from Rails 4.2.0/Ruby 2.2.1 to Rails 4.2.1/Ruby 2.2.2 I get the following error when attempting to access a secret from `Rails.application.secrets` in an initializer or environment configuration file:

```
NameError: uninitialized constant Rails::Application::YAML
~/apps/foo/vendor/bundle/ruby/2.2.0/gems/railties-4.2.1/lib/rails/application.rb:391:in `secrets'
~/apps/foo/config/initializers/foo.rb:1:in `<top (required)>'
~/apps/foo/vendor/bundle/ruby/2.2.0/gems/railties-4.2.1/lib/rails/engine.rb:652:in `block in load_config_initializer'
~/apps/foo/vendor/bundle/ruby/2.2.0/gems/activesupport-4.2.1/lib/active_support/notifications.rb:166:in `instrument'
~/apps/foo/vendor/bundle/ruby/2.2.0/gems/railties-4.2.1/lib/rails/engine.rb:651:in `load_config_initializer'
~/apps/foo/vendor/bundle/ruby/2.2.0/gems/railties-4.2.1/lib/rails/engine.rb:616:in `block (2 levels) in <class:Engine>'
~/apps/foo/vendor/bundle/ruby/2.2.0/gems/railties-4.2.1/lib/rails/engine.rb:615:in `each'
~/apps/foo/vendor/bundle/ruby/2.2.0/gems/railties-4.2.1/lib/rails/engine.rb:615:in `block in <class:Engine>'
~/apps/foo/vendor/bundle/ruby/2.2.0/gems/railties-4.2.1/lib/rails/initializable.rb:30:in `instance_exec'
~/apps/foo/vendor/bundle/ruby/2.2.0/gems/railties-4.2.1/lib/rails/initializable.rb:30:in `run'
~/apps/foo/vendor/bundle/ruby/2.2.0/gems/railties-4.2.1/lib/rails/initializable.rb:55:in `block in run_initializers'
~/apps/foo/vendor/bundle/ruby/2.2.0/gems/railties-4.2.1/lib/rails/initializable.rb:44:in `each'
~/apps/foo/vendor/bundle/ruby/2.2.0/gems/railties-4.2.1/lib/rails/initializable.rb:44:in `tsort_each_child'
~/apps/foo/vendor/bundle/ruby/2.2.0/gems/railties-4.2.1/lib/rails/initializable.rb:54:in `run_initializers'
~/apps/foo/vendor/bundle/ruby/2.2.0/gems/railties-4.2.1/lib/rails/application.rb:352:in `initialize!'
~/apps/foo/config/environment.rb:5:in `<top (required)>'
~/apps/foo/vendor/bundle/ruby/2.2.0/gems/railties-4.2.1/lib/rails/application.rb:328:in `require'
~/apps/foo/vendor/bundle/ruby/2.2.0/gems/railties-4.2.1/lib/rails/application.rb:328:in `require_environment!'
~/apps/foo/vendor/bundle/ruby/2.2.0/gems/railties-4.2.1/lib/rails/application.rb:457:in `block in run_tasks_blocks'
```

Adding `require('yaml')` in the initializer or environment configuration solves the problem, but YAML should already be being loaded by Rails.

The problem is exactly the same as described in [this blog post](http://blog.tristanmedia.com/2014/04/rails-4-1-initializers-and-secrets-yml/) regarding Rails 4.1.

**Comment by pixeltrix** [comment]
This fixed in 64dacdd689ba3f3699f43d2971bbeb3a9d7195e9 - thanks for your report. This was working before because sprockets was requiring yaml but since version 3 it no longer does - did you do a `bundle update` for everything? That would probably have bumped the sprockets to version 3 as we don't lock it to version 2.

**Comment by codyrobbins** [comment]
Yup, I did a `bundle update` and went to Sprockets 3 that’s when things went awry. Thanks for the quick response!
